3scale/porta
# Log in with Auth tokens

<img width="1017" alt="Screenshot 2020-05-06 at 11 55 13" src="https://user-images.githubusercontent.com/4183971/81164332-7806de00-8f90-11ea-9832-e8b72279ca7a.png">



**What this PR does / why we need it**:

We need a way to communicate with Porta API to work on the real implementation and not leaving the page components using the OAS examples. This is not a production ready AuthN solution, just a temporary way of login and secure the calls.

**Which issue(s) this PR fixes** 

fixes https://issues.redhat.com/browse/THREESCALE-4992

**Verification steps** 

1. checkout the branch
2. `cd portafly`
3. `yarn install`
4. `yarn start`
5. Open browser `http://localhost:3003` and you should hit the log in page


Note: This steps will be evolving when introducing new features.

**Special notes for your reviewer**:

- Since we are not sure how we are going to store the access token just yet, and not even know how we are going to get it, as a temporary measure it'll be stored in `localStorage` (not recommended because it could be target of a XSS and leaked). However, if in the future we work with JWT, it might be Ok since it can't be forged, changing a JWT payload invalidates the JWT as a whole and auth servers won't accept them.

- ~~There's a possibility to "check" the validity of the token with a call to the API, then store it or show an error message.~~ NO

- ~~Should "private" pages links be hidden in navigation? Or should the entire navigation be hidden and we should show only a login page?~~ Not logged in -> login page, Logged in -> App

**TODO**
